CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
+
+project(rapidfuzz CXX)
+include(GNUInstallDirs)
+
+option(BUILD_TESTS "Build unit tests" OFF)
+option(BUILD_STATIC "Build static library" ON)
+option(BUILD_SHARED "Build shared library" ON)
+
+set(SOURCES
+	rapidfuzz/fuzz.txx
+	rapidfuzz/levenshtein.txx
+	rapidfuzz/process.txx
+	rapidfuzz/utils.txx
+)
+
+set(HEADERS
+	rapidfuzz/fuzz.hpp
+	rapidfuzz/levenshtein.hpp
+	rapidfuzz/process.hpp
+	rapidfuzz/utils.hpp
+	rapidfuzz/details/SplittedSentenceView.hpp
+	rapidfuzz/details/type_traits.hpp
+	rapidfuzz/details/types.hpp
+	rapidfuzz/details/unicode.hpp
+)
+
+function(set_rapidfuzz_properties TARGET)
+# provide a namespaced alias for clients to 'link' against if RapidFuzz is included as a sub-project
+add_library(rapidfuzz::${TARGET} ALIAS ${TARGET})
+
+set_target_properties(${TARGET} PROPERTIES VERSION 0.0.0
+                                PUBLIC_HEADER "${HEADERS}"
+                                LINKER_LANGUAGE CXX
+                                CXX_STANDARD 17)
```

Comment:
ok I'll look into that too, I experienced some weird error this afternoon regarding the lack of std::sort and std::find_if which I workaround by upgrading C++ standard
I'll add an option to let the use choose the standard

Reply:
Hm C++14 definetly works on a lot of compilers, since thats what I build the pythpn version with. C++11 I did only build with gcc10 so far.

Reply:
I think the error I got yesterday was due to the mess of the static / shared lib, by setting the library as interface I can build my test program with C++11
also this `CXX_STANDARD` option doesn't exist for `INTERFACE` target so I removed it.
The user could set its preferred C++ version in his CMakeLists.txt.
I don't know how to specify a minimum required version.

Reply:
I think with CMake 3.8 we could use
```
target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_11)
```
for this (for previous version it would be possible to select it based on features)
https://crascit.com/2015/03/28/enabling-cxx11-in-cmake/

Reply:
It seems to also work with CMake 2.8. 
At least setting `cmake_minimum_required(VERSION 2.8)` on my test project and adding `target_compile_features(rapidfuzz INTERFACE cxx_std_11)` works.

Reply:
see 7786060b